sdk/core/Azure.Core.TestFramework/src/RecordedTestBase.cs

```diff
@@ -513,6 +520,79 @@ public static Task Delay(RecordedTestMode mode, int milliseconds = 1000, int? pl
             return Task.CompletedTask;
         }
 
+        private void AssertTestProxyToolIsInstalled()
+        {
+            if (s_ranTestProxyValidation ||
+                TestEnvironment.GlobalIsRunningInCI ||
+                !TestEnvironment.IsWindows ||
+                TestEnvironment.DisableTestProxyPrompt ||
+                AssetsJsonPath == null)
+            {
+                return;
+            }
+
+            lock (s_syncLock)
+            {
+                if (s_ranTestProxyValidation)
+                {
+                    return;
+                }
+
+                s_ranTestProxyValidation = true;
+
+                try
+                {
+                    if (IsTestProxyToolInstalled(global: true) || IsTestProxyToolInstalled(global: false))
+                    {
+                        return;
+                    }
+
+                    string path = Path.Combine(
+                        TestEnvironment.RepositoryRoot,
+                        "eng",
+                        "scripts",
+                        "Install-TestProxyTool.ps1");
+
+                    var processInfo = new ProcessStartInfo("pwsh.exe", path)
+                    {
+                        UseShellExecute = true
+                    };
+
+                    var process = Process.Start(processInfo);
+
+                    if (process != null)
+                    {
+                        process.WaitForExit();
+                    }
+                }
+                catch (Exception)
+                {
+                    // Ignore
+                }
+            }
+        }
+
+        private bool IsTestProxyToolInstalled(bool global)
+        {
+            string args = "tool list " + (global ? "--global" : "--local");
```

Comment:
I'm not sure that we need the local support. All of our docs recommend installing globally. If someone has installed it locally, they should still just install it globally as well.